xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1566,6 +1567,15 @@ bool CPVRClients::OpenStream(const CPVRRecordingPtr &channel)
       client->OpenStream(channel))
   {
     CSingleLock lock(m_critSection);
+
+    CDateTime endTime = channel->RecordingTimeAsLocalTime() +
+        CDateTimeSpan(0, 0, channel->GetDuration() / 60, channel->GetDuration() % 60);
+
+    m_bIsRecordingInProgress = (endTime > CDateTime::GetCurrentDateTime());
```

Comment:
this is an assumption that does not hold for all cases. time setting may be wrong or switch to/from DST has happened.

backend needs to be queried to get this right.